python/pyiceberg/manifest.py

```diff
@@ -163,6 +194,70 @@ def __repr__(self) -> str:
 )
 
 
+@singledispatch
+def partition_field_to_data_file_partition_field(partition_field_type: IcebergType) -> PrimitiveType:
+    raise TypeError(f"Unsupported partition field type: {partition_field_type}")
+
+
+@partition_field_to_data_file_partition_field.register(LongType)
+@partition_field_to_data_file_partition_field.register(DateType)
+@partition_field_to_data_file_partition_field.register(TimeType)
+@partition_field_to_data_file_partition_field.register(TimestampType)
+@partition_field_to_data_file_partition_field.register(TimestamptzType)
+def _(partition_field_type: PrimitiveType) -> IntegerType:
+    return IntegerType()
```

Comment:
The `partition_types` got from `PartitionSpec` contains the type of the field in the table schema. Some of these are different from the actual types in `data_file.partition`. For example, 
```json
{
	"name": "partition",
	"type": {
		"type": "record",
		"name": "r102",
		"fields": [{
			"name": "tpep_pickup_datetime_day",
			"type": ["null", {
				"type": "int",
				"logicalType": "date"
			}],
			"default": null,
			"field-id": 1000
		}]
	},
	"field-id": 102
}
```
Hence, we need some measure to convert these to the right types for Avro Writer